config/vufind/Blender.ini

```diff
@@ -0,0 +1,226 @@
+; Blender configuration. Note that all backends enabled here must have their own
+; configuration set up properly first, i.e. VuFind must be configured to use each
+; individual backend first before Blender can be used.
+
+; Enabled backends and their labels. A label is a translatable string that is
+; displayed in the results list for each result. Use an empty string for no label.
+; Backend order defines the order the results are fetched from each backend.
+[Backends]
+Solr = "Local Index"
+Primo = "CDI"
+EDS = "EDS"
+
+[Blending]
```

Comment:
Maybe I'm just missing/misunderstanding something, but I'm seeing strange results when I use this default configuration without a working Primo. (I get the same results whether I use the config as-is, with Primo failing, or when I comment out Primo in the [Backends] section above). I would expect the result order on the first page of 20 records to be:

Solr x 2, EDS x 2, Solr x 7, EDS x 7, Solr x 2

(Based on the "initialResults" followed by the round robin setting).

What I'm actually getting is:

Solr x 2, EDS x 2, Solr x 1, EDS x 7, Solr x 8

Is something weird happening when initialResults doesn't get what it's expecting? If I comment out the two `initialResults[] = Primo` lines, I still don't get what I expect, but it does change things. I end up with:

Solr x 2, EDS x 2, Solr x 3, EDS x 7, Solr x 6

I haven't studied code at all, but just thought I'd pass long these results in case they are helpful.

Reply:
It was broken, now fixed. The blocks start from beginning regardless of initialResults, so what you should be getting is:

Solr x 2, EDS x 2, Solr x 3, EDS x 7, Solr 6

This can be revisited, but this is simpler as you don't have to take what's been boosted and what's available into account when determining the source for a position later on.